src/Kiota.Builder/KiotaBuilder.cs

```diff
@@ -1106,6 +1107,23 @@ private CodeParameter GetIndexerParameterType(OpenApiUrlTreeNode currentNode, Op
         };
         return result;
     }
+    private static IDictionary<string, OpenApiPathItem> GetPathItems(OpenApiUrlTreeNode currentNode, bool validateIsParamterNode = true)
+    {
+        if((!validateIsParameterNode || currentNode.IsParameter) && currentNode.PathItems.Any())
+        {
+            return currentNode.PathItems;
+        }
+
+        if (currentNode.Children.Any())
+        {
+            return currentNode.Children
+                .SelectMany(static x => GetPathItems(x, false))
+                .DistinctBy(static x => x.Key, StringComparer.Ordinal)
+                .ToDictionary(static x => x.Key, static x => x.Value, StringComparer.Ordinal);
+        }
+
+        return new Dictionary<string, OpenApiPathItem>();
```

Comment:
Can you also replace that by https://learn.microsoft.com/en-us/dotnet/api/system.collections.immutable.immutabledictionary-2.empty?view=net-8.0

So we avoid unnecessary allocations please? (you'll probably need to add a using statement as well)